crates/matrix-sdk/src/client.rs

```diff
@@ -630,6 +636,49 @@ impl Client {
         self
     }
 
+    /// Add an arbitrary value for use as event handler context.
+    ///
+    /// The value can be obtained in an event handler by adding an argument of
+    /// the type [`Ctx<T>`][crate::event_handler::Ctx].
+    ///
+    /// If a value of the same type has been added before, it will be
+    /// overwritten.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
```

Comment:
Any reason why we wouldn't want to run this? Usually we don't run things if the example needs to make a HTTP request.

Reply:
Well, the `async` block wouldn't be executed anyways 🤷🏼

Reply:
Should I wrap the whole thing in a `#[tokio::test] async fn`? I just did the same thing as on `register_event_handler` since I didn't think running it would do worthwhile, and added `no_run` so the test suite doesn't spend time spawning a process (or thread, or whatever) for nothing.

Reply:
Well I don't know, generally I like running doc examples since they essentially become tests that way. 

While it might not be important here, other people tend to copy things. Adding a `no_run` where it's not needed increases the chance that we'll have more examples that aren't being run.

> Should I wrap the whole thing in a #[tokio::test] async fn?

Other examples just use `block_on()`, isn't that the more lightweight approach here?

Reply:
Oh another reason why it's good to run doc examples, it forces you to write an example that works, i.e. people can copy the example and expect it to not crash.

Reply:
There's not much possibility for things to crash here, but I'll look at the `block_on`s you mentioned and update this and the `register_event_handler` test 👍🏼